integration-healthcheck/src/main/java/org/symphonyoss/integration/healthcheck/AsyncCompositeHealthIndicator.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2016-2017 Symphony Integrations - Symphony LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.symphonyoss.integration.healthcheck;
+
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthAggregator;
+import org.springframework.boot.actuate.health.HealthIndicator;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * {@link HealthIndicator} that returns health indications from all registered delegates using
+ * asynchronous calls.
+ * Created by rsanchez on 16/01/17.
+ */
+public class AsyncCompositeHealthIndicator implements HealthIndicator {
+
+  /**
+   * Thread pool size
+   */
+  private static final Integer MAX_POOL_SIZE = 10;
+
+  /**
+   * Timeout to verify if the execution was done
+   */
+  private static final Long EXECUTION_TIMEOUT = 1000L;
+
+  /**
+   * Registered indicators
+   */
+  private final Map<String, HealthIndicator> indicators;
+
+  /**
+   * Health aggregator
+   */
+  private final HealthAggregator healthAggregator;
+
+  public AsyncCompositeHealthIndicator(HealthAggregator aggregator) {
+    this.healthAggregator = aggregator;
+    this.indicators = new HashMap<>();
+  }
+
+  /**
+   * Register new indicator
+   * @param name Health indicator name
+   * @param indicator Health indicator object
+   */
+  public void addHealthIndicator(String name, HealthIndicator indicator) {
+    this.indicators.put(name, indicator);
+  }
+
+  @Override
+  public Health health() {
+    try {
+      Map<String, Future<Health>> result = asyncExecution();
+      Map<String, Health> healths = extractResult(result);
+      return this.healthAggregator.aggregate(healths);
+    } catch (InterruptedException e) {
+      return Health.down().withException(e).build();
```

Comment:
I mustn't interrupt the request thread because it will fail the request execution and won't provide the response to the caller.

I changed the code to log the exception and define a more descriptive error message.